pkg/intermediate/aggregate.go

```diff
@@ -79,24 +80,29 @@ func (a *aggregation) AggregateMsgByFlowKey(message *entities.Message) error {
 		if err != nil {
 			return err
 		}
-		a.correlateRecords(flowKey, record)
+		a.correlateRecords(*flowKey, record)
 	}
 	return nil
 }
 
 func (a *aggregation) GetFlowKeyRecordMap() map[FlowKey][]entities.Record {
-	return a.flowKeyRecordMap
+	a.flowKeyRecordLock.RLock()
+	defer a.flowKeyRecordLock.RUnlock()
+	resMap := make(map[FlowKey][]entities.Record)
```

Comment:
We may consider returning a pointer of the new map to avoid making a second copy during returned-by-value?

Reply:
[Map](https://blog.golang.org/maps) in Golang is reference type like pointer, so it should be returned by reference.

Reply:
Cool, nice to learn that. Thanks.

Reply:
The main objective of this function is to iterate over the records and perform a common operation.
Could we do something like below in Antrea to avoid the step of creating a copy of the map? I think this map will be common to the cluster w.r.t flow aggregator implementation and could be very large; making a copy can involve high overhead.

https://github.com/vmware-tanzu/antrea/blob/master/pkg/agent/flowexporter/connections/connections.go#L209

Reply:
I see the above function is currently utilized only in tests, and all the others are internal functions that access the map directly. However, exposing this function may lead to inefficient usage.

Reply:
Thanks for the suggestion. For flow aggregator, we will need to get the map, obtain the records and send them, that is the use case for exposing the function.

Reply:
Checked with @dreamtalen that passing a callback function will be enough for flow aggregator to process the records.